test/tests/performance/pbs_multi_sched_perf.py

```diff
@@ -0,0 +1,169 @@
+            self.server.manager(MGR_CMD_CREATE, SCHED,
+                                a, id=sched_name)
+            self.scheds[sched_name].create_scheduler()
+            self.scheds[sched_name].start()
+            self.server.manager(MGR_CMD_SET, SCHED,
+                                {'scheduling': 'True'}, id=sched_name)
+
+    def setup_queues_nodes(self):
+        a = {'queue_type': 'execution',
+             'started': 'True',
+             'enabled': 'True'}
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq1')
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq2')
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq3')
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq4')
+        self.server.manager(MGR_CMD_CREATE, QUEUE, a, id='wq5')
+        p1 = {'partition': 'P1'}
+        self.server.manager(MGR_CMD_SET, QUEUE, p1, id='wq1')
+        p2 = {'partition': 'P2'}
+        self.server.manager(MGR_CMD_SET, QUEUE, p2, id='wq2')
+        p3 = {'partition': 'P3'}
+        self.server.manager(MGR_CMD_SET, QUEUE, p3, id='wq3')
+        p4 = {'partition': 'P4'}
+        self.server.manager(MGR_CMD_SET, QUEUE, p4, id='wq4')
+        p5 = {'partition': 'P5'}
+        self.server.manager(MGR_CMD_SET, QUEUE, p5, id='wq5')
+        node = str(self.mom.shortname)
+        self.server.manager(MGR_CMD_SET, NODE, p1, id=node + '[0]')
+        self.server.manager(MGR_CMD_SET, NODE, p2, id=node + '[1]')
+        self.server.manager(MGR_CMD_SET, NODE, p3, id=node + '[2]')
+        self.server.manager(MGR_CMD_SET, NODE, p4, id=node + '[3]')
+        self.server.manager(MGR_CMD_SET, NODE, p5, id=node + '[4]')
+
+    def submit_jobs(self, num_jobs=1, attrs=None, user=TEST_USER):
+        """
+        Submit num_jobs number of jobs with attrs attributes for user.
+        Return a list of job ids
+        """
+        if attrs is None:
+            attrs = {ATTR_q: 'workq'}
+        ret_jids = []
+        for _ in range(num_jobs):
+            J = Job(user, attrs)
+            jid = self.server.submit(J)
+            ret_jids += [jid]
+
+        return ret_jids
+
+    @timeout(3600)
+    def test_multi_sched_perf(self):
+        """
+        Test time taken to schedule and run 5k jobs with
+        single scheduler and workload divided among 5 schedulers.
+        """
+        a = {'resources_available.ncpus': 1000}
+        self.server.create_vnodes(self.mom.shortname, a, 5, self.mom)
+        a = {'scheduling': 'False'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        self.submit_jobs(5000)
+        start = time.time()
+        a = {'scheduling': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a)
+        self.server.expect(SCHED, {'state': 'scheduling'},
+                           id='default', max_attempts=10)
+        self.scheduler.log_match(
+            "Starting Scheduling Cycle", starttime=int(start),
+            max_attempts=30)
+        a = {'scheduling': 'False'}
```

Comment:
Here you can use self.scheduler.run_scheduling_cycle().  This will handle turning on and off scheduling to run a cycle and wait for the cycle to finish.  You can't use it below because you'd end up running the multischeds in a serial fashion.